File: python/ast_dump.py
import ast
import sys

if __name__ == '__main__':

    args = sys.argv

    if len(args) < 2:
        # not found : analyze target (*.py)
        print("not found : analyze target (*.py)")
        exit(1)

    target = args[1]

    # is python file? (ended with ".py"?)
    if target[-3:] != ".py":
        print("target suffix is not '.py'")
        exit(1)

    with open(args[1], "r") as f:
        tree = ast.parse(f.read())
        print(ast.dump(tree))

    exit(0)

File: ruby/ast_dump.rb
if ARGV.size == 0 then
    pp "not found : analyze target (*.rb)"
    exit(1)
end

if ARGV[0].end_with?(".rb") == false then
    pp "target suffix is not '.rb'"
    exit(1)
end

pp RubyVM::AbstractSyntaxTree.parse_file(ARGV[0])
exit(0)
